weblate/addons/webhooks.py

```diff
@@ -0,0 +1,76 @@
+# Copyright © Michal Čihař <michal@weblate.org>
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+from django.utils import timezone
+from django.utils.translation import gettext_lazy
+
+from weblate.addons.base import ChangeBaseAddon
+from weblate.addons.forms import WebhooksAddonForm
+from weblate.api.tasks import webhook_delivery_task
+from weblate.trans.util import split_plural
+
+
+class WebhookAddon(ChangeBaseAddon):
+    name = "weblate.webhook.webhook"
+    # TODO: improve description
+    verbose = gettext_lazy("Webhooks")
+    description = gettext_lazy("some desc")
+
+    settings_form = WebhooksAddonForm
+    # TODO: find a webhook icon
+
+    def change_event(self, change) -> None:
+        config = self.instance.configuration
+        if change.action in config["events"]:
+            payload = self.build_webhook_payload(change)
+            webhook_delivery_task.delay(
+                url=config["webhook_url"],
+                headers=self.build_headers(change, payload),
+                data=payload,
+            )
+
+    def build_webhook_payload(self, change) -> dict:
```

Comment:
I'd really prefer to base this on the JSON schema so that the payload content is formally defined. Maybe reusing the Fedora Messaging schema will be the way to go, but I really didn't investigate.